pkg/bitcoin/electrum/electrum.go

```diff
@@ -301,7 +301,34 @@ func (c *Connection) GetBlockHeader(
 	return blockHeader, nil
 }
 
-// GetTransactionsForPublicKeyHash get confirmed transactions that pays the
+// GetTransactionMerkleProof gets the Merkle proof for a given transaction.
+// The transaction's hash and the block the transaction was included in the
+// blockchain need to be provided.
+func (c *Connection) GetTransactionMerkleProof(
+	transactionHash bitcoin.Hash,
+	blockHeight uint,
+) (*bitcoin.TransactionMerkleProof, error) {
+	getMerkleProofResult, err := requestWithRetry(
+		c,
+		func(
+			ctx context.Context,
+			client *electrum.Client,
+		) (*electrum.GetMerkleProofResult, error) {
+			return client.GetMerkleProof(
+				ctx,
+				transactionHash.String(),
+				uint32(blockHeight),
+			)
+		},
+	)
```

Comment:
We should pass the transaction hash in the ReversedByteOrder and be consistent with other functions implementations, see an example in the `GetTransaction` function.

```suggestion
    txID := transactionHash.Hex(bitcoin.ReversedByteOrder)
    
	getMerkleProofResult, err := requestWithRetry(
		c,
		func(
			ctx context.Context,
			client *electrum.Client,
		) (*electrum.GetMerkleProofResult, error) {
			return client.GetMerkleProof(
				ctx,
				txID,
				uint32(blockHeight),
			)
		},
	)
```

Reply:
Done in c91b4932fd967bd22767e424b18845ff7dd211d5.